### PR TITLE
Validations and some other things

### PR DESCRIPTION
**Pull request by HaydarAk**
Some minor additions for the Configuration Model, mainly idsm:validations.

### CODE REVIEW

File: model/communication/Endpoint.ttl [hunk 1]
```diff
@@ -51,6 +51,14 @@ ids:AppEndpoint
     # rdfs:subClassOf ids:Endpoint ; #TODO: not possible. See explanation above.
     rdfs:label "DataApp Endpoint"@en ;
     rdfs:comment "Description of endpoints a DataApp offers."@en ;
+    idsm:validation [
+        idsm:forProperty ids:appEndpointPort;
+        idsm:constraint idsm:NotNull ;
```
Comment (JohannesLipp) [inline model/communication/Endpoint.ttl:56]: The port declaration could be optional, as some might use default ports. You choose

File: model/communication/Proxy.ttl [hunk 2]
```diff
@@ -18,6 +18,10 @@ ids:Proxy a owl:Class;
         idsm:forProperty ids:proxyURI;
         idsm:constraint idsm:NotNull;
                   ];
+    idsm:validation [
+        idsm:forProperty ids:proxyAuthentication;
```
Comment (JohannesLipp) [inline model/communication/Proxy.ttl:22]: Please verify, if we need to enforce auth for any proxy. This might be optional.

Comment (JohannesLipp) [reply model/communication/Proxy.ttl:22]: For userAuthentication below, we allow `idsm:OneToMany`. Could this be an option here?

File: codes/RouteDeployMethod.ttl [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+
+# Instances
+# ---------
+
+# Logical operators
+idsc:APACHE_CAMEL a ids:RouteDeployMethod;
+    rdfs:label "Apache Camel"@en ;
+    rdfs:comment "Indicates that an AppRoute is deployed / implemented using Apache Camel."@en ;
+    .
```
[the remark this answers is not in the document]
Comment (jlangkau) [reply codes/RouteDeployMethod.ttl:14]: Yes. I do it (very often) this way, too:

### EOF

File: model/communication/Endpoint.ttl (hunk 1 above)
Comment (HaydarAk) [reply model/communication/Endpoint.ttl:56]: Good point! Havent thought about that.

File: model/communication/Proxy.ttl (hunk 2 above)
Comment (HaydarAk) [reply model/communication/Proxy.ttl:22]: You're right. A proxy could be used without authentication!

**Comment by HaydarAk** [comment]
Thanks @jlangkau  and @JohannesLipp for your help.
Added your recommendations in the last commit.

Additionally, as @jlangkau highlighted (in another channel), we should stay technology-agonistic for the routing related things too. Therefore I removed the Apache Camel instance and changed the property to range string for these kind of information.

**Pull request merged**